From 07d4224f2e4821dc6dcc4a3f5d41da6563751564 Mon Sep 17 00:00:00 2001
From: Yaguang Tang <tangyg2@lenovo.com>
Date: Fri, 20 Oct 2017 17:28:32 +0800
Subject: [PATCH 47/91] Remove the wrap for skip inspect rbd disk info

Libvirt already support get rbd disk info since 2.0, so we can remove this
wrap waring.

Change-Id: Ie11f64ce5dd9ce60b574ef1f6445d60e60b1887b
---
 ceilometer/compute/virt/libvirt/inspector.py       | 34 +++++++++-------------
 .../unit/compute/virt/libvirt/test_inspector.py    |  2 +-
 2 files changed, 14 insertions(+), 22 deletions(-)

diff --git a/ceilometer/compute/virt/libvirt/inspector.py b/ceilometer/compute/virt/libvirt/inspector.py
index e92170a..d3dfa63 100644
--- a/ceilometer/compute/virt/libvirt/inspector.py
+++ b/ceilometer/compute/virt/libvirt/inspector.py
@@ -144,27 +144,19 @@ class LibvirtInspector(virt_inspector.Inspector):
         domain = self._get_domain_not_shut_off_or_raise(instance)
         tree = etree.fromstring(domain.XMLDesc(0))
         for disk in tree.findall('devices/disk'):
-            disk_type = disk.get('type')
-            if disk_type:
-                if disk_type == 'network':
-                    LOG.warning(
-                        'Inspection disk usage of network disk '
-                        '%(instance_uuid)s unsupported by libvirt' % {
-                            'instance_uuid': instance.id})
-                    continue
-                # NOTE(lhx): "cdrom" device associated to the configdrive
-                # no longer has a "source" element. Releated bug:
-                # https://bugs.launchpad.net/ceilometer/+bug/1622718
-                if disk.find('source') is None:
-                    continue
-                target = disk.find('target')
-                device = target.get('dev')
-                if device:
-                    block_info = domain.blockInfo(device)
-                    yield virt_inspector.DiskInfo(device=device,
-                                                  capacity=block_info[0],
-                                                  allocation=block_info[1],
-                                                  physical=block_info[2])
+            # NOTE(lhx): "cdrom" device associated to the configdrive
+            # no longer has a "source" element. Releated bug:
+            # https://bugs.launchpad.net/ceilometer/+bug/1622718
+            if disk.find('source') is None:
+                continue
+            target = disk.find('target')
+            device = target.get('dev')
+            if device:
+                block_info = domain.blockInfo(device)
+                yield virt_inspector.DiskInfo(device=device,
+                                              capacity=block_info[0],
+                                              allocation=block_info[1],
+                                              physical=block_info[2])
 
     @libvirt_utils.raise_nodata_if_unsupported
     @libvirt_utils.retry_on_disconnect
diff --git a/ceilometer/tests/unit/compute/virt/libvirt/test_inspector.py b/ceilometer/tests/unit/compute/virt/libvirt/test_inspector.py
index bb36ef1..46325d1 100644
--- a/ceilometer/tests/unit/compute/virt/libvirt/test_inspector.py
+++ b/ceilometer/tests/unit/compute/virt/libvirt/test_inspector.py
@@ -373,7 +373,7 @@ class TestLibvirtInspection(base.BaseTestCase):
         with mock.patch('ceilometer.compute.virt.libvirt.utils.'
                         'refresh_libvirt_connection', return_value=conn):
             disks = list(self.inspector.inspect_disk_info(self.instance, None))
-            self.assertEqual(0, len(disks))
+            self.assertEqual(1, len(disks))
 
     def test_inspect_disk_info_without_source_element(self):
         dom_xml = """
-- 
2.7.4

